Assets/Scripts/Tools/ExtendedRandom.cs

```diff
@@ -0,0 +1,144 @@
+        }
+
+        public RandomX(int seed)
+        {
+            random = new System.Random(seed);
+        }
+
+        /// <summary>
+        /// Returns a random point inside or on a circle with radius 1.0
+        /// </summary>
+        public Vector2 InsideUnitCircle()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi]
+            float radius = Value(); // Generate radius between [0, 1]
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: radius * Mathf.Cos(angle),
+                y: radius * Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// Returns a random point on the edge of a circle with radius 1.0
+        /// </summary>
+        public Vector2 OnUnitCircle()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi]
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: Mathf.Cos(angle),
+                y: Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// Returns a random point inside or on a sphere with radius 1.0
+        /// </summary>
+        public Vector3 InsideUnitSphere()
+        {
+            float phi = Value(0, 2) * Mathf.PI; // Generates angle between [0, 2*Pi]
+            float theta = Value() * Mathf.PI; // Generates angle between [0, Pi]
+            float radius = Value(); // Generates radius between [0, 1]
+
+            // Convert from spherical coordinates to regular coordinates
+            return new Vector3(
+                x: radius * Mathf.Sin(theta) * Mathf.Cos(phi),
+                y: radius * Mathf.Sin(theta) * Mathf.Sin(phi),
+                z: radius * Mathf.Cos(theta));
+        }
+
+        /// <summary>
+        /// Returns a random point on the surface of a sphere with radius 1.0
+        /// </summary>
+        public Vector3 OnUnitSphere()
+        {
+            float phi = Value(0, 2) * Mathf.PI; // Generates angle between [0, 2*Pi]
+            float theta = Value() * Mathf.PI; // Generates angle between [0, Pi]
+
+            // Convert from spherical coordinates to regular coordinates
+            return new Vector3(
+                x: Mathf.Sin(theta) * Mathf.Cos(phi),
+                y: Mathf.Sin(theta) * Mathf.Sin(phi),
+                z: Mathf.Cos(theta));
+        }
+
+        /// <summary>
+        /// Returns a random rotation 
+        /// </summary>
+        public Quaternion Rotation()
+        {
+            float x = Value();
+            float y = Value();
+            float z = Value();
+            float w = Value();
+
+            return new Quaternion(x, y, z, w).normalized;
+        }
+
+        /// <summary>
+        /// Returns a random float within [0, 1]
+        /// </summary>
+        public float Value()
+        {
+            return (float)random.NextDouble();
```

Comment:
Random.NextDouble() has exclusive max, should be in description and ranges that depend on it. 
https://learn.microsoft.com/en-us/dotnet/api/system.random.nextdouble

Reply:
fixat